### ISSUE TITLE
AttributeError when importing smc module

### ISSUE THREAD

**Issue by entomologe** [Forcepoint/fp-NGFW-SMC-python#46]
Hello,

since recently I get the following error when importing "smc"; I don't remember changing anything apart from general updating via "apt upgrade". I even tried with a fresh user "test":

```
$ cat test.py
#!/usr/bin/env python3
import smc
```
```
$ ./test.py 
Traceback (most recent call last):
  File "/home/test/ngfw/./test.py", line 2, in <module>
    import smc
  File "/home/test/.local/lib/python3.10/site-packages/smc/__init__.py", line 2, in <module>
    import smc.api.session
  File "/home/test/.local/lib/python3.10/site-packages/smc/api/session.py", line 9, in <module>
    from smc.api.entry_point import Resource
  File "/home/test/.local/lib/python3.10/site-packages/smc/api/entry_point.py", line 6, in <module>
    from smc.base.structs import SerializedIterable
  File "/home/test/.local/lib/python3.10/site-packages/smc/base/structs.py", line 126, in <module>
    class NestedDict(collections.MutableMapping): 
AttributeError: module 'collections' has no attribute 'MutableMapping'
```
```
$ cat /etc/os-release 
PRETTY_NAME="Ubuntu 22.10"
NAME="Ubuntu"
VERSION_ID="22.10"
VERSION="22.10 (Kinetic Kudu)"
VERSION_CODENAME=kinetic
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=kinetic
LOGO=ubuntu-logo
```
```
$ dpkg --list python3-pip python3
...
ii  python3        3.10.6-1             amd64        interactive high-level object-oriented language (default python3 version)
ii  python3-pip    22.2+dfsg-1ubuntu0.2 all          Python package installer
````

I tried reinstalling the package via `pip3 install --force-reinstall smc-python` without success. 

Kind thanks for any idea how I got there and how to repair it.

**Comment by entomologe** [comment]
Hello Greg,

> Which version of smc-python are you using please?

$ pip3 show smc-python
Name: smc-python
Version: 0.6.2
...

> Please note that library name is not anymore **smc-python** but **fp-NGFW-SMC-python**.

Thank you for your clarification. Meanwhile I'm running into other problems using this library using [session.login](https://fp-ngfw-smc-python.readthedocs.io/en/latest/pages/session.html#creating-the-session) (URL, API_KEY, API_VERSION are python variables in my case):
```
>>> session.login(url=URL, api_key=API_KEY, api_version=API_VERSION)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/user/.local/lib/python3.10/site-packages/smc/api/session.py", line 558, in login
    self._session = self._get_session(request)
  File "/home/user/.local/lib/python3.10/site-packages/smc/api/session.py", line 622, in _get_session
    retry = Retry(
TypeError: Retry.__init__() got an unexpected keyword argument 'method_whitelist'
```

Here are the supposedly relevant packages that are installed:
```
$ pip3 list | egrep -i '(urllib3|smc|urllib|requests)'
fp-NGFW-SMC-python     1.0.19
requests               2.30.0
requests-toolbelt      0.9.1
requests-unixsocket    0.3.0
smc-python-monitoring  1.2.0
urllib3                2.0.2
```

**Comment by gregory83** [comment]
For me it is related to python version. I'm checking internally. I'll come back one we do have an answer.

**Comment by entomologe** [comment]
> For me it is related to python version. I'm checking internally. I'll come back one we do have an answer.

Thanks for your fast reply. Which python version are you using so I may try with those then?

**Comment by gregory83** [comment]
From what I can see:
* 3.10.11
* 3.8.10
* 3.9.16

**Comment by entomologe** [comment]
While using python3.11 the problem disappeared.
`$ python3.11 -m pip install fp-NGFW-SMC-python requests smc-python-monitoring urllib3 websocket-client`

So the hint with the python version was very helpful, kind thanks.

Should the issue be resolved now?

**Comment by lilianValeroFp** [comment]
Perfect! So we can close the issue.

**Comment by m0canu1** [comment]
Hi, I also have the same issue.

I have the fp-NGFW-SMC-python-1.0.21 and tried with these Python versions:

- 3.7.9
- 3.8.10
- 3.9.13
- 3.10.11
- 3.11.3

but none of them worked.


> ```
> >>> session.login(url=URL, api_key=API_KEY, api_version=API_VERSION)
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
>   File "/home/user/.local/lib/python3.10/site-packages/smc/api/session.py", line 558, in login
>     self._session = self._get_session(request)
>   File "/home/user/.local/lib/python3.10/site-packages/smc/api/session.py", line 622, in _get_session
>     retry = Retry(
> TypeError: Retry.__init__() got an unexpected keyword argument 'method_whitelist'
> ```

Do you have any suggestions?

**Comment by thomasdevulder** [comment]
Hello,

This issue will be fixed in version 1.0.22 which will be released tomorrow
